src/librustdoc/clean/inline.rs

```diff
@@ -372,6 +370,12 @@ pub fn build_impl(
             if !cx.renderinfo.borrow().access_levels.is_public(did) {
                 return;
             }
+
+            if let Some(stab) = tcx.lookup_stability(did) {
+                if stab.level.is_unstable() && stab.feature == sym::rustc_private {
+                    return;
+                }
+            }
```

Comment:
Hmm, why is this bit necessary? Shouldn't all these cases be covered by line 349 above?

Reply:
the above diff is for `associated_trait`, this is for `for_`.

Reply:
Right, but `for_` is the same trait, just in a rustdoc data structure. I guess theoretically they could be different if there's a bug, but in that case I think it should panic, not silently ignore it.

Reply:
I'm not sure but we also have is_public check for both.

Reply:
Oh I see ... well, this is a weird setup then but no need to fix it here.